include/zephyr/sys_clock.h

```diff
@@ -289,6 +289,19 @@ static inline bool sys_timepoint_expired(k_timepoint_t timepoint)
 	return K_TIMEOUT_EQ(sys_timepoint_timeout(timepoint), Z_TIMEOUT_NO_WAIT);
 }
 
+/**
+ * @brief Compare two timepoint values.
+ *
+ * This function is used to compare two timepoint values and find the one that is pointing
+ * to earlier point in time. It does not verify if given timepoints are already expired.
+ *
+ * @param a Timepoint to compare
+ * @param b Timepoint to compare against.
+ * @return zero if both timepoints are the same. Negative value if timepoint A points to earlier
+ * time, positive otherwise.
+ */
+int sys_timepoint_cmp(k_timepoint_t a, k_timepoint_t b);
```

Comment:
This function needs an implementation if `CONFIG_SYS_CLOCK_EXISTS==n`.

The implementation is small and trivial, maybe turn this into a `static inline` function (for both `CONFIG_SYS_CLOCK_EXISTS` modes?

Reply:
Oh.. I see. I did not notice the `timeout.c` is not always compiled in.

I think it could be `static inline` as it is so trivial.